spec/grid_spec.rb

```diff
@@ -1,16 +1,75 @@
 require_relative "../lib/grid"
 
 RSpec.describe Grid do
-  describe "#mark" do
-    it "updates the given cell" do
+  describe "#mark_computer" do
+    it "assigns O to the given coordinates" do
       grid = Grid.new
 
-      grid.mark(coordinates: "A1", value: "X")
+      grid.mark_computer(coordinates: "A1")
+
+      expect(grid.value("A1")).to eq "O"
+    end
+  end
+
+  describe "#mark_player" do
+    it "assigns X to the given coordinates" do
+      grid = Grid.new
+
+      grid.mark_player(coordinates: "A1")
 
       expect(grid.value("A1")).to eq "X"
     end
   end
 
+  describe "#random_coordinates" do
+    it "returns random coordinates from the available ones" do
+      srand(0)
+      grid = Grid.new("A1" => "X", "B2" => "O", "C3" => "X")
+
+      expect(grid.random_coordinates).to eq "C1"
+    end
+
+    it "is nil when all coordinates have been used" do
+      grid = Grid.new(
+        "A1" => "X",
+        "A2" => "X",
+        "A3" => "X",
+        "B1" => "O",
+        "B2" => "O",
+        "B3" => "O",
+        "C1" => "X",
+        "C2" => "X",
+        "C3" => "X"
+      )
+
+      expect(grid.random_coordinates).to be_nil
+    end
+  end
+
+  describe "#valid_coordinates?" do
+    it "is true when the value is in the grid" do
+      expect(Grid.new.valid_coordinates?("A1")).to be true
+      expect(Grid.new.valid_coordinates?("B2")).to be true
+      expect(Grid.new.valid_coordinates?("C3")).to be true
+    end
+
+    it "is false when in a row before the first row of the grid" do
```

Comment:
Didn't notice these before, but the descriptions don't make a lot of sense. 

This one looks like `is false when the the coordinates don't match the pattern`

Reply:
Yeah, it's trying to be symmetrical and cover the case before the rows start. I picked `@` because it's the ASCII character before `A`, but I agree it doesn't really match the description. Updating this to be about format, and not having a spec fore "before," seems appropriate.